language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -670,15 +664,56 @@ impl DependencyGraph {
         let manifest = parse_move_manifest_from_file(&pkg_path)
             .with_context(|| format!("Parsing manifest for '{}'", name))?;
 
-        let kind = pkg.kind.clone();
-        self.extend_graph(
-            &kind,
+        let sub_graph = DependencyGraph::new(
```

Comment:
My thinking was that we'd want to a "disjoint" graph for comparison rather than the entire "merged" one. Wasn't sure it would make a difference but was erring on the side of caution.